sql/src/test/java/org/apache/druid/quidem/DruidQuidemTestBase.java

```diff
@@ -0,0 +1,207 @@
+
+/**
+ * Execute Quidem tests in Druid.
+ *
+ * How these tests work:
+ * <ol>
+ * <li>Test cases are in .iq files - contract of these files is that they
+ * produce themselves if it was executed without errors</li>
+ * <li>Executor (this class) picks up these files and runs them as part of unit
+ * testruns</li>
+ * <li>System under test is connected via an adapter which looks like a JDBC
+ * driver</li>
+ * </ol>
+ *
+ * Example usage:
+ * <ol>
+ * <li>Write new .iq test as a under the appropriate directory; with command for
+ * expectations but without specifying them.</li>
+ * <li>Run the test - it will produce a ".iq.out" file next to the ".iq"
+ * one.</li>
+ * <li>Copy over the .iq.out to .iq to accept the changes</li>
+ * </ol>
+ *
+ * To shorten the above 2 steps
+ *
+ */
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public abstract class DruidQuidemTestBase
+{
+
+  public static final String IQ_SUFFIX = ".iq";
+  /**
+   * System property name for "overwrite mode"; note: empty value is treated as
+   * true
+   */
+  private static final String OVERWRITE_PROPERTY = "quidem.overwrite";
+
+  private static final String PROPERTY_FILTER = "quidem.filter";
+
+  private FileFilter filter = TrueFileFilter.INSTANCE;
+
+  private DruidQuidemRunner druidQuidemRunner;
+
+  public DruidQuidemTestBase()
+  {
+    String filterStr = System.getProperty(PROPERTY_FILTER, null);
+    if (filterStr != null) {
+      if (!filterStr.endsWith("*") && !filterStr.endsWith(IQ_SUFFIX)) {
+        filterStr = filterStr + IQ_SUFFIX;
+      }
+      filter = new WildcardFileFilter(filterStr);
+    }
+    druidQuidemRunner = new DruidQuidemRunner();
+  }
+
+  /** Creates a command handler. */
+  protected CommandHandler createCommandHandler()
+  {
+    return Quidem.EMPTY_COMMAND_HANDLER;
+  }
+
+  @ParameterizedTest
+  @MethodSource("getFileNames")
+  public void test(String testFileName) throws Exception
+  {
+    File inFile = new File(getTestRoot(), testFileName);
+
+    final File outFile = new File(inFile.getParentFile(), inFile.getName() + ".out");
+    druidQuidemRunner.run(inFile, outFile);
+  }
+
+  public static class DruidQuidemRunner
+  {
+    public DruidQuidemRunner()
+    {
+    }
+
+    public void run(File inFile) throws Exception
+    {
+      File outFile = new File(inFile.getParent(), inFile.getName() + ".out");
+      run(inFile, outFile);
+    }
+
+    public void run(File inFile, final File outFile) throws Exception
+    {
+      FileUtils.mkdirp(outFile.getParentFile());
+      try (Reader reader = Util.reader(inFile);
+          Writer writer = Util.printWriter(outFile);
+          Closer closer = new Closer()) {
+
+        DruidQuidemConnectionFactory connectionFactory = new DruidQuidemConnectionFactory();
+        ConfigBuilder configBuilder = Quidem.configBuilder()
+            .withConnectionFactory(connectionFactory)
+            // this is not nice - but it makes it possible to do queryContext
+            // changes
```

Comment:
A nit about the comment, "this is not nice" will make the next developer that comes along believe that there is something wrong with the code.  But, like, have no clue what is wrong.  Like, what's "not nice"?  Why is it "not nice"?  Either leave out the statement and just say "we include the connectionFactory again so that it can see queryContext changes" or update the comment to include an explanation of what's sub-optimal and what future work could be done to make it better.

Sorry for being anal about "just a comment" but when developers are reading new code, they tend to put a ton of faith in the comments written in that code, even though the comments are often an after-thought/just a random thing written and forgotten by the original developer.

Reply:
decided that its the best to remove this comment - as it doesn't add much value; I believe its not a big issue